lambda_src/drivers/destination_s3.py

```diff
@@ -101,23 +99,31 @@ def write(
     row_index: int,
 ) -> Dict[Text, Any]:
     bucket, prefix = parse_destination_uri(destination)
-    encoded_datum = (
-        '\n'.join(json.dumps(d) for d in datum)
-        if isinstance(datum, list)
-        else json.dumps(datum, default=str)
-    )
+    if isinstance(datum, bytes):
+        encoded_datum = datum
+        encoded_datum_hash = sha256(encoded_datum).hexdigest()
+        prefixed_filename = lazy_format(prefix, hash=lambda: encoded_datum_hash)
```

Comment:
Used `encoded_datum_hash` in the lambda to not re-calculate the hash on line [125](https://github.com/Snowflake-Labs/geff/pull/64/files#diff-361e1d3c9eb36aa3cc3dfcb657f34db3c71d7b22a6f5b3d10fb2a37a3bfd81faR125)